cmd/subnetcmd/publish.go

```diff
@@ -0,0 +1,380 @@
+					if !yes {
+						ux.Logger.PrintToUser("User canceled, nothing got published.")
+						return nil
+					}
+				}
+			} else {
+				aliases := make([]string, len(matches))
+				for i, a := range matches {
+					aliases[i] = a.Name()
+				}
+				alias, err = app.Prompt.CaptureList("Pick an alias", aliases)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// ask for a new alias
+func getNewAlias() (string, error) {
+	return app.Prompt.CaptureString("Please provide an alias for the repository we are going to use")
+}
+
+func getRepoURL(reposDir string) error {
+	if repoURL == "" {
+		path := filepath.Join(reposDir, alias)
+		repo, err := git.PlainOpen(path)
+		if err != nil {
+			// TODO: probably a debug log; the alias might not have been created yet,
+			// so the repo might not exist, therefore just ignore it!
+		} else {
+			// there is a repo already for this alias, let's try to figure out the remote URL from there
+			conf, err := repo.Config()
+			if err != nil {
+				// TODO Would we really want to abort here?
+				return err
+			}
+			remotes := make([]string, len(conf.Remotes))
+			i := 0
+			for _, r := range conf.Remotes {
+				// TODO: Usually there's only one URL for a remote, but more are supported - should we too?
+				remotes[i] = r.URLs[0]
+				i++
+			}
+			repoURL, err = app.Prompt.CaptureList("Which is the remote URL for this repo?", remotes)
+			if err != nil {
+				// TODO Would we really want to abort here?
+				return err
+			}
+			return nil
+		}
+		repoURL, err = app.Prompt.CaptureString("Please provide the repository URL")
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// loadYAMLFile loads a YAML file from disk into a concrete types.Definition object
+// using generics. It's role really is solely to verify that the YAML content is valid.
+func loadYAMLFile[T types.Definition](path string, defType T) error {
+	fileBytes, err := os.ReadFile(path)
+	if err != nil {
+		return err
+	}
+	return yaml.Unmarshal(fileBytes, &defType)
+}
+
+func getSubnetInfo(sc *models.Sidecar) (*types.Subnet, error) {
+	homepage, err := app.Prompt.CaptureEmpty("What is the homepage of the Subnet project?", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	desc, err := app.Prompt.CaptureEmpty("Please provide a free-text description of the Subnet", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	maintrs, canceled, err := app.Prompt.CaptureListDecision(
+		app.Prompt,
+		"Who are the maintainers of the Subnet?",
+		app.Prompt.CaptureEmail,
+		"Provide a maintainer",
+		"Maintainer",
+		"",
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+	if canceled {
+		ux.Logger.PrintToUser("Publishing aborted")
+		return nil, errors.New("canceled by user")
+	}
+
+	strMaintrs := make([]string, len(maintrs))
+	for i, m := range maintrs {
+		strMaintrs[i] = m.(string)
+	}
+
+	// TODO: In this version, we are publishing 1 subnet and exactly 1 VM.
+	// 1. Will the VM list has to be editale in the future?
```

Comment:
No, it should be a property of the subnet config. At this point, the subnet should be "locked-in" and deployed to fuji or prod.

Reply:
Well, actually I could see a situation where another chain is added later. Theoretically possible, but probably not worth worrying about right now.

Reply:
OK. Removing all this and setting the `sidecar.Name` for VM